### DISCUSSION TITLE
Validation attribute name

### DISCUSSION THREAD

**Discussion by haztakki** [filamentphp/filament#3518] [Help]
Hello **again**,
Taking this example straight from the documentation:

```
TextInput::make('slug')->rules([
    function () {
        return function (string $attribute, $value, Closure $fail) {
            if ($value === 'foo') {
                $fail("The {$attribute} is invalid.");
            }
        };
    },
])
```

How would I go about setting the attribute name? Chaining the name() method has no effect.

 The data.validation.3537e93b-58b2-4694-8ea3-afe32ba0d3d2.value is invalid. 
 The data.validation.75b2d449-3735-4b3f-afa9-fb9876606cdc.value is invalid.

**Comment by danharrin** [comment]
I believe that we have a `->validationAttribute()` method for all form inputs? Else, we use the `->label()`

**Reply by haztakki** [reply]
Hey @danharrin,
Thanks for your quick reply. I can't seem to get any of those to work. I am chaining it to a text input inside a repeater. The name is still dynamically generated.

```
                        TextInput::make(name: 'value')
                            // ...simplified
                            ->label('value') // this
                            ->validationAttribute('value') // or this
                            ->rules([
                                function () {
                                    return function (string $attribute, $value, Closure $fail) {
                                        if ($value === 'foo') {
                                            $fail("The {$attribute} is invalid.");
                                        }
                                    };
                                },
                            ]),
```

**Reply by danharrin** [reply]
Very weird, please open an issue with a reproduction repository and ill look into it

**Reply by haztakki** [reply]
Done. It is most likely me doing some stupid. #3520
